File: converter/length.go
package converter

import (
	"fmt"
	"strconv"
)

func LenthConverter(valueStr, from, to string) (string, error) {
	value, err := strconv.ParseFloat(valueStr, 64)
	if err != nil {

		return "", err
	}

	// convert to meter first
	meter := convertToMeter(value, from)

	// convert to target uint
	result := convertFromMeter(meter, to)
	formattedResult := fmt.Sprintf("%.2f", result)
	return formattedResult, nil
}

func convertToMeter(val float64, unit string) float64 {
	switch unit {
	case "mm":
		return val / 1000
	case "cm":
		return val / 100
	case "m":
		return val
	case "km":
		return val * 1000
	case "inch":
		return val * 0.0254
	case "foot":
		return val * 0.3048
	case "yard":
		return val * 0.9144
	case "mile":
		return val * 1609.34
	default:
		return val
	}
}

func convertFromMeter(meters float64, unit string) float64 {
	switch unit {
	case "mm":
		return meters * 1000
	case "cm":
		return meters * 100
	case "m":
		return meters
	case "km":
		return meters / 1000
	case "inch":
		return meters / 0.0254
	case "foot":
		return meters / 0.3048
	case "yard":
		return meters / 0.9144
	case "mile":
		return meters / 1609.34
	default:
		return meters
	}
}

File: static/js/length.js
 function convert() {
            const value = parseFloat(document.getElementById('value').value);
            const from = document.getElementById('from').value;
            const to = document.getElementById('to').value;

            if (!value || isNaN(value)) {
                document.getElementById('result-value').textContent = "0";
                return;
            }

            // Convert to meter first
            let meters = value;
            switch(from) {
                case "mm": meters = value / 1000; break;
                case "cm": meters = value / 100; break;
                case "m": meters = value; break;
                case "km": meters = value * 1000; break;
                case "inch": meters = value * 0.0254; break;
                case "foot": meters = value * 0.3048; break;
                case "yard": meters = value * 0.9144; break;
                case "mile": meters = value * 1609.34; break;
            }

            // Convert meter to target unit
            let result = meters;
            switch(to) {
                case "mm": result = meters * 1000; break;
                case "cm": result = meters * 100; break;
                case "m": result = meters; break;
                case "km": result = meters / 1000; break;
                case "inch": result = meters / 0.0254; break;
                case "foot": result = meters / 0.3048; break;
                case "yard": result = meters / 0.9144; break;
                case "mile": result = meters / 1609.34; break;
            }

            document.getElementById('result-value').textContent = result.toFixed(4);
        }